src/python-fastui/fastui/forms.py

```diff
@@ -194,10 +193,10 @@ def unflatten(form_data: ds.FormData) -> NestedDict:
                 d[part] = {}
             d = d[part]
 
-        if len(values) == 1:
-            d[last_key] = values[0]
-        else:
+        if (last_key in model.model_fields and _t.get_origin(model.model_fields[last_key].annotation) is list) or len(values) > 1:
```

Comment:
this isn't going to work for lots of other types e.g. tuple, set, frozenset.

Instead of introspecting the python type, I'd rather we iterated over the JSON Schema and worked out which fields should take multiple values that way.

With that we can also:
* move the logic out of `unflatten` thus separating concerns
* cache the result per model
* have a more general solution as JSON Schema already simplifies many types into an array schema

Since we already have logic to flatten JSON Schema, we might be able to use that and therefore pass a simple `dict[key, bool]` to this method.